spec/models/validations/sales/setup_validations_spec.rb

```diff
@@ -6,43 +6,100 @@
 
   describe "#validate_saledate" do
-    context "when saledate is blank" do
-      let(:record) { FactoryBot.build(:sales_log, saledate: nil) }
+    context "with sales_in_crossover_period == false" do
+      context "when saledate is blank" do
+        let(:record) { build(:sales_log, saledate: nil) }
 
-      it "does not add an error" do
-        setup_validator.validate_saledate(record)
+        it "does not add an error" do
+          setup_validator.validate_saledate(record)
 
-        expect(record.errors).to be_empty
+          expect(record.errors).to be_empty
+        end
       end
-    end
 
-    context "when saledate is in the 22/23 financial year" do
-      let(:record) { FactoryBot.build(:sales_log, saledate: Time.zone.local(2023, 1, 1)) }
+      context "when saledate is in the 22/23 financial year" do
+        let(:record) { build(:sales_log, saledate: Time.zone.local(2023, 1, 1)) }
 
-      it "does not add an error" do
-        setup_validator.validate_saledate(record)
+        it "does not add an error" do
+          setup_validator.validate_saledate(record)
 
-        expect(record.errors).to be_empty
+          expect(record.errors).to be_empty
+        end
+      end
+
+      context "when saledate is before the 22/23 financial year" do
+        let(:record) { build(:sales_log, saledate: Time.zone.local(2020, 1, 1)) }
+
+        it "adds error" do
+          setup_validator.validate_saledate(record)
+
+          expect(record.errors[:saledate]).to include("Enter a date within the 22/23 financial year, which is between 1st April 2022 and 31st March 2023")
+        end
       end
-    end
 
-    context "when saledate is before the 22/23 financial year" do
-      let(:record) { FactoryBot.build(:sales_log, saledate: Time.zone.local(2022, 1, 1)) }
+      context "when saledate is after the 22/23 financial year" do
+        let(:record) { build(:sales_log, saledate: Time.zone.local(2025, 4, 1)) }
 
-      it "adds error" do
-        setup_validator.validate_saledate(record)
+        it "adds error" do
+          setup_validator.validate_saledate(record)
 
-        expect(record.errors[:saledate]).to include(I18n.t("validations.setup.saledate.financial_year"))
+          expect(record.errors[:saledate]).to include("Enter a date within the 22/23 financial year, which is between 1st April 2022 and 31st March 2023")
+        end
       end
     end
 
-    context "when saledate is after the 22/23 financial year" do
-      let(:record) { FactoryBot.build(:sales_log, saledate: Time.zone.local(2023, 4, 1)) }
+    context "with sales_in_crossover_period == true" do
+      # before do
+      #   allow(FormHandler.instance).to receive(:sales_in_crossover_period?).and_return(false)
+      # end
+
+      around do |example|
+        Timecop.freeze(Time.zone.local(2022, 5, 1)) do
+          Singleton.__init__(FormHandler)
+          example.run
+        end
+        Timecop.return
+        Singleton.__init__(FormHandler)
+      end
```

Comment:
These tests might start failing when we go into the next year, because we will no longer have the 21/22 form so `sales_in_crossover_period?` will return false

Maybe there's a way you can set those dates dynamically, or at least test with the next years crossover date?